integrations/lambda/demo.ipynb

```diff
@@ -1098,12 +1099,54 @@
     "\n",
     "# Attach Policy to the IAM Role\n",
     "policy_arn = \"arn:aws:iam::aws:policy/AmazonTimestreamFullAccess\"\n",
-    "iam_client.attach_role_policy(\n",
-    "    RoleName=role_name,\n",
-    "    PolicyArn=policy_arn\n",
-    ")\n",
     "\n",
-    "print(f\"Attached Timestream write policy to {role_name}\")\n",
+    "timestream_write_policy = {\n",
+    "    \"Version\": \"2012-10-17\",\n",
+    "    \"Statement\": [\n",
+    "        {\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Action\": [\n",
+    "                \"timestream:WriteRecords\",\n",
+    "                \"timestream:Select\",\n",
+    "                \"timestream:DescribeTable\",\n",
+    "\t\t\t\t\"timestream:CreateTable\"\n",
+    "            ],\n",
+    "            \"Resource\": f\"arn:aws:timestream:{lambda_client.meta.region_name}:{account_id}:database/{DATABASE_NAME}/table/{TABLE_NAME}\"\n",
+    "        },\n",
+    "        {\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Action\": [\n",
+    "                \"timestream:DescribeEndpoints\"\n",
+    "            ],\n",
+    "            \"Resource\": \"*\"\n",
+    "        },\n",
+    "        {\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Action\": [\n",
+    "                \"timestream:DescribeDatabase\",\n",
+    "\t\t\t\t\"timestream:CreateDatabase\"\n",
+    "            ],\n",
+    "            \"Resource\": f\"arn:aws:timestream:{lambda_client.meta.region_name}:{account_id}:database/{DATABASE_NAME}\"\n",
+    "        }\n",
+    "    ]\n",
+    "}\n",
+    "\n",
+    "# Add the Timestream write policy to the role\n",
+    "try:\n",
+    "    iam_client.put_role_policy(\n",
+    "        RoleName=role_name,\n",
+    "        PolicyName='TimestreamSampleWritePolicy',\n",
+    "        PolicyDocument=json.dumps(timestream_write_policy)\n",
+    "    )\n",
+    "    print(f\"Attached TimestreamSampleWritePolicy policy to role: {role_name}\")\n",
+    "except Exception as e:\n",
+    "    print(f\"Error attaching TimestreamSampleWritePolicy: {e}\")\n",
+    "\n",
+    "print(f\"Attached TimestreamSampleWritePolicy policy to {role_name}\")\n",
+    "\n",
+    "# A wait period is necessary between role creation and Lambda creation\n",
+    "# in order to avoid a \"The role defined for the function cannot be assumed by Lambda\" error.\n",
+    "time.sleep(7)\n",
```

Comment:
I have added retries with exponential backoffs, both for when creating the Lambda function and for sending data to the Lambda function, since errors can happen in both cases when resources have been newly created.